cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesExport.java

```diff
@@ -64,182 +66,437 @@ public List<Parameter> getParameters() {
+     */
+    private void updateDependenciesIGM(Map<CgmesSubset, CgmesMetadataModel> subsetModels, String boundaryEqId, String boundaryTpId) {
+        // Retrieve EQ model ID
+        String eqModelId = subsetModels.get(CgmesSubset.EQUIPMENT).getId();
+        if (eqModelId == null || eqModelId.isEmpty()) {
+            return;
+        }
+
+        // TP and SSH depend on EQ
+        subsetModels.get(CgmesSubset.TOPOLOGY).addDependentOn(eqModelId);
+        subsetModels.get(CgmesSubset.STEADY_STATE_HYPOTHESIS).addDependentOn(eqModelId);
+
+        // SV depends on TP and SSH
+        subsetModels.get(CgmesSubset.STATE_VARIABLES)
+                .addDependentOn(subsetModels.get(CgmesSubset.TOPOLOGY).getId())
+                .addDependentOn(subsetModels.get(CgmesSubset.STEADY_STATE_HYPOTHESIS).getId());
+
+        // EQ depends on EQ_BD (if present)
+        if (boundaryEqId != null) {
+            subsetModels.get(CgmesSubset.EQUIPMENT).addDependentOn(boundaryEqId);
+        }
+
+        // SV depends on TP_BD (if present)
+        if (boundaryTpId != null) {
+            subsetModels.get(CgmesSubset.STATE_VARIABLES).addDependentOn(boundaryTpId);
+        }
+    }
+
+    /**
+     * Check that the given network is consistent with a CGM export.
+     * @param network The network to export as a CGM.
+     * @param context The context used by the export.
+     */
+    private void checkCgmConsistency(Network network, CgmesExportContext context) {
+        // Check that the network has subnetworks
+        if (network.getSubnetworks().size() < 2) {
+            LOG.error("Network {} must have at least 2 subnetworks for a CGM export.", network.getId());
+        }
+
+        // QoCDC, rule CgmSvSshVersionMismatch: SSHs and SV must have same scenario time
+        ZonedDateTime scenarioTime = network.getCaseDate();
+        for (Network subnetwork : network.getSubnetworks()) {
+            if (!subnetwork.getCaseDate().equals(scenarioTime)) {
+                LOG.error("Parent network doesn't have the same scenarioTime as subnetwork: {}.", subnetwork.getId());
             }
-            if (profiles.contains("SSH")) {
-                try (OutputStream out = new BufferedOutputStream(ds.newOutputStream(filenameSsh, false))) {
-                    XMLStreamWriter writer = XmlUtil.initializeWriter(true, INDENT, out);
-                    SteadyStateHypothesisExport.write(network, writer, context);
-                }
-            } else {
-                saveLegacyIdsFromPropertiesForSvDependencies(network, CgmesSubset.STEADY_STATE_HYPOTHESIS, context);
+        }
+
+        // QoCDC, rule CgmSvSshVersionMismatch: SSHs and SV must have same version
+        // If version has been provided as a parameter, use it, otherwise take the max version of updated models
+        if (context.getModelVersion() == null || context.getModelVersion().isEmpty()) {
+            int maxVersion = maxOfVersionAndValue(network, CgmesSubset.STATE_VARIABLES, 2);
```

Comment:
In case of cgm export with Ssh initial version 0, the exported model will have version = 2 instead of 1. Is this normal ?

Reply:
Fixed. 

Looking at QoCDC it is not clear if the first version of IGMs should be 0 or 1. Lines 654, 655 of version 3.3.1 read as the following: _The fileVersion is exactly three characters long positive integer number between 000 and 999, i.e.,
the first positive integer is 001 and the last 999_, and the tables 3 and 4 use 000 (table 3) or 001 (table 4) for the submitted IGM filename examples.

Considering this, the proposed fix is the following:

If no version number has been read (either from original models or parameters), it is assumed everything was at version 1, so next version is 2. (It seems less dangerous to put all files at version 2 and eventually leave a hole than write version 1 and have a potential implicit collision).

If input data is explicitly given as version 0, exported version will be 1.